File: app/src/main/java/com/example/advancedprayertimes/ui/PrayerSettingsViewPagerAdapter.java
package com.example.advancedprayertimes.UI;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.advancedprayertimes.Logic.Enums.EPrayerTimeType;
import com.example.advancedprayertimes.UI.Fragments.TabPrayerBeginningSettingsFragment;
import com.example.advancedprayertimes.UI.Fragments.TabPrayerEndSettingsFragment;
import com.example.advancedprayertimes.UI.Fragments.TabPrayerSpecificSettingsFragment;

public class PrayerSettingsViewPagerAdapter extends FragmentStateAdapter
{
    private EPrayerTimeType _prayerType;

    public PrayerSettingsViewPagerAdapter(FragmentActivity fragmentActivity, EPrayerTimeType prayerType)
    {
        super(fragmentActivity);
        this._prayerType = prayerType;
    }

    @Override
    public Fragment createFragment(int position)
    {
        switch(position)
        {
            case 0:
                return TabPrayerBeginningSettingsFragment.newInstance(_prayerType);
            case 1:
                return TabPrayerEndSettingsFragment.newInstance(_prayerType);
            case 2:
                return TabPrayerSpecificSettingsFragment.newInstance(_prayerType);

            default:
                return null;
        }
    }

    @Override
    public int getItemCount()
    {
        if(_prayerType == EPrayerTimeType.Asr || _prayerType == EPrayerTimeType.Isha)
        {
            return 3;
        }

        return 2;
    }
}

File: app/src/main/kotlin/com/example/advancedprayertimes/ui/prayer_setting_ui_components/PrayerSettingsViewPagerAdapter.kt
package com.example.advancedprayertimes.ui.prayer_setting_ui_components

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.example.advancedprayertimes.logic.enums.EPrayerTimeType
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.advancedprayertimes.ui.prayer_setting_ui_components.tab_fragments.PrayerBeginningSettingsTabFragment
import com.example.advancedprayertimes.ui.prayer_setting_ui_components.tab_fragments.PrayerEndSettingsTabFragment
import com.example.advancedprayertimes.ui.prayer_setting_ui_components.tab_fragments.PrayerSpecificSettingsTabFragment

class PrayerSettingsViewPagerAdapter(fragmentActivity: FragmentActivity?, private val _prayerType: EPrayerTimeType
) : FragmentStateAdapter(
    fragmentActivity!!
) {

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> PrayerBeginningSettingsTabFragment.newInstance(_prayerType)
            1 -> PrayerEndSettingsTabFragment.newInstance(_prayerType)
            else -> PrayerSpecificSettingsTabFragment.newInstance(_prayerType)
        }
    }

    override fun getItemCount(): Int {
        return when(_prayerType) {
            EPrayerTimeType.Asr, EPrayerTimeType.Maghrib, EPrayerTimeType.Isha -> 3
            else -> 2
        }
    }
}